### PR TITLE
AV-885: predicate-based recategorization system for sixodp_harvester

### PR DESCRIPTION
**Pull request by bzar**
- Created tiny category mapping language
- Replace direct category mapping with new rules

### CODE REVIEW

File: modules/ckanext-ytp_main/ckanext/ytp/harvesters/sixodp_harvester.py [hunk 1]
```diff
@@ -62,21 +47,57 @@ def parse_datetime(datetime_string):
...
+    def _and(*predicates):
+        def f(values):
+            return all(_evaluate(p, values) for p in predicates)
+        return f
 
-        if mapped_groups is not None:
-            for mapped_group in mapped_groups:
-                log.info("Mapping Group %s => %s", group, mapped_group)
-                groups.append({'name': mapped_group})
-        else:
```
Comment (Aaronkala) [inline modules/ckanext-ytp_main/ckanext/ytp/harvesters/sixodp_harvester.py]: The old implementation returned the group if it was not included in the map, but the new implmentation only returns groups that are in the map. Is this intended? Will it result in issues?

**Comment by Aaronkala** [comment]
It would be great to have test for functions like this, because they require some time to understand and can easily be broken if not understood correctly

File: modules/ckanext-ytp_main/ckanext/ytp/harvesters/sixodp_harvester.py (hunk 1 above)
Comment (Aaronkala) [reply modules/ckanext-ytp_main/ckanext/ytp/harvesters/sixodp_harvester.py]: The returning lines are under this actually :) (76, 77)

Comment (bzar) [reply modules/ckanext-ytp_main/ckanext/ytp/harvesters/sixodp_harvester.py]: Intended. We know specifically what groups we want to map into what and how. The previous implementation was meant to only optionally map some categories into others while defaulting to passthrough.

**Comment by bzar** [comment]
Added some comments to the code to make it easier to reason about.

**Pull request merged**